### ISSUE TITLE
File streaming not ending

### ISSUE THREAD

**Issue by imachug** [HelloZeroNet/ZeroNet#1232]
### Step 1: Please describe your environment

  * ZeroNet version: 0.6.0 (rev 3178)
  * Operating system: Windows 10
  * Web browser: Chrome
  * Tor status: disabled
  * Opened port: yes
  * Special configuration: none

### Step 2: Describe the problem:

Ranged requests like `0-100000` when file size is `100` only never end. Moreover, after a couple of such requests even normal requests fail.

#### Steps to reproduce:

  1. Create new site
  2. Copy file `00changelog.d` and copy it to site root: [00changelog.zip](https://github.com/HelloZeroNet/ZeroNet/files/1595079/00changelog.zip)
  3. Open DevTools
  4. Define function:
```javascript
function ajax(size) {
	page.cmd("wrapperGetAjaxKey", [], ajax => {
		let xhr = new XMLHttpRequest;
		xhr.open("GET", "00changelog.d?ajax_key=" + ajax);
		xhr.responseType = "arraybuffer";
		xhr.setRequestHeader("Range", "bytes=0-" + size);
		xhr.onload = e => {
			if(xhr.status != 206) {
				console.error(xhr.status);
			} else {
				console.log("Loaded");
			}
		};
		xhr.onreadystatechange = () => {
			console.log("State", xhr.readyState);
		};
		xhr.onerror = e => {
			console.error(e);
		};
		console.log("Begin loading");
		xhr.send(null);
	});
}
```
Basically this will send a range request.

Now most important:
1. Send `ajax(50)`
2. Send `ajax(1000)` a couple of times (about 5 is enough for me)
3. Send `ajax(50)`

#### Observed Results:

1.
```
Begin loading
State 2
State 3
Loaded
```
2.
```
Begin loading
State 2
State 3
```
3.
```
Begin loading
State 2
State 3
```

Notice that data was sent (`State 3`), though even normal request didn't end (`Loaded`).

#### Expected Results:

1.
```
Begin loading
State 2
State 3
Loaded
```
2.
```
Begin loading
State 2
State 3
Loaded
```
3.
```
Begin loading
State 2
State 3
Loaded
```

**Comment by imachug** [comment]
@HelloZeroNet I could create a PR for this, but I don't understand why this happens at all.

**Comment by HelloZeroNet** [comment]
Thanks for reporting, it should work fine now: https://github.com/HelloZeroNet/ZeroNet/commit/3fb9f900f6c0d08abd8facc987af4a274d31d56b
